lib/hawkular/base_client.rb

```diff
@@ -175,6 +178,16 @@ def tenant_header
       headers
     end
 
+    # @!visibility private
+    def connect_error(fault)
+      if fault.is_a?(SocketError)
+        HawkularConnectionException.new(fault.to_s)
+      elsif [Errno::ECONNREFUSED, Errno::ECONNRESET, Errno::EHOSTUNREACH,
```

Comment:
I was looking for a way to not have to, but IIRC the parent class of all of these is StandardError. It seems that needs to be handled this way.